File: Backend/ArcadeX.Persistence/Entities/Review.cs
namespace ArcadeX.Persistence.Entities;

public class Review
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public Guid GameId { get; set; }
    public int Rating { get; set; }
    public string? Comment { get; set; }
    public DateTime CreatedAt { get; set; }
    public User User { get; set; } = null!;
    public Game Game { get; set; } = null!;
    public ICollection<ReviewComment> ReviewComments { get; set; } = new List<ReviewComment>();

}

File: Frontend/src/app/feactures/interfaces/private/Review.ts
export interface IReview {
  id: string;
  gameId: string;
  userId?: string;
  rating: number;
  comment?: string;
  createdAt?: string;
}

export interface ICreateReview {
  gameId: string;
  rating: number;
  comment?: string;
}

export interface IUpdateReview {
  rating: number;
  comment?: string;
}
